internal/sqlitestore/store.go

```diff
@@ -0,0 +1,246 @@
+package sqlitestore
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/jinzhu/gorm"
+	"github.com/writhe/kudosaurus"
+
+	// NOTE: Needed for Sqlite3 support
+	_ "github.com/jinzhu/gorm/dialects/sqlite"
+)
+
+// SqliteStore - implements interface DataSource
+type SqliteStore struct {
```

Comment:
I like the fact that you've separated SQL functions from rest of the app. That's a good practice.

Reply:
The idea was to be able to swap out Sqlite for another data source.

Reply:
The best part is that when you swap sqlite for sql or psql, you don't even have to change that code (only imports) 🏅

Reply:
you actually have to change this line 👉  `db, err := gorm.Open("sqlite3", s.path)`